### ISSUE TITLE
Does distributed model support directly get_preds()?

### ISSUE THREAD

**Issue by f-sky** [fastai/fastai#1935]
I launch a distributed training referring https://docs.fast.ai/distributed.html#prepare-your-script 
I notice that during training, training set is split to different gpus and thus reduces iterations on each gpu.
But when I directly load model and run "get_preds()", I notice that iterations on each gpu does not reduce, that is to say, each gpu run "get_preds()" on full validation set.
In the following, I describe experiment results.
In all experiments, num_train=50000,num_val=10000,num_gpus=4,batch_size=128.
1. train and then get_preds
```python
from fastai.vision import *
from fastai.vision.models.wrn import wrn_22
from fastai.distributed import *
from fastai.callbacks import SaveModelCallback
import argparse
parser = argparse.ArgumentParser()
parser.add_argument("--local_rank", type=int)
args = parser.parse_args()
torch.cuda.set_device(args.local_rank)
torch.distributed.init_process_group(backend='nccl', init_method='env://')

path = untar_data(URLs.CIFAR)
ds_tfms = ([*rand_pad(4, 32), flip_lr(p=0.5)], [])
data = ImageDataBunch.from_folder(path, valid='test', ds_tfms=ds_tfms, bs=128).normalize(cifar_stats)
learn = Learner(data, wrn_22(), metrics=accuracy).to_distributed(args.local_rank)
learn.fit_one_cycle(1, 3e-3, wd=0.4, div_factor=10, pct_start=0.5)
learn.get_preds()
```
Fitting takes 97 iterations on training set and 20 iters on validation set.
get_preds() takes 20 iters on validation set.
2. load and then get_preds
```python
...
learn = Learner(data, wrn_22(), metrics=accuracy).to_distributed(args.local_rank)
learn.load('best_model')
learn.get_preds()
```
get_preds() takes 79 iters on validation set.
3. load, train and get_preds
```python
...
learn = Learner(data, wrn_22(), metrics=accuracy).to_distributed(args.local_rank)
learn.load('best_model')
learn.get_preds()
learn.fit_one_cycle(1, 3e-3, wd=0.4, div_factor=10, pct_start=0.5)
learn.get_preds()
```
The first get_preds() takes 79 iters while the second get_preds() takes 20 iters.

It seems that without fitting firstly, get_preds() will not launch distributed. Is it a bug? 
Now I want to get_preds() with only loading but not fitting, how to use multi-gpus?

**Comment by sgugger** [comment]
The distributed callback only changes the dataloaders to make them distributed during the `on_train_begin` event, that's why you see the difference. The bug is that at the end of training, the dataloaders aren't put back to what they were, I'll fix that today (afterward you will always see 79 iters in `get_preds`)

Note that `get_preds` , isn't written to support distributed training: it doesn't all_reduce the losses and metrics so each GPU will only have its metrics. I'd advise to use it with non-splitted DataLoaders if you don't write your custom evaluation function.

**Comment by sgugger** [comment]
Done now, closing this issue. You can add a request for a new feature (a function that properly does `get_preds` for distributed) on the [forum](https://forums.fast.ai/) and maybe tackle it and submit a PR :)
